components/theia/packages/gitpod-extension/src/browser/cli-service-client.ts

```diff
@@ -25,28 +25,27 @@ export interface CliServiceClient extends TheiaCLIService {
 @injectable()
 export class CliServiceClientImpl implements CliServiceClient {
 
+    private readonly workspaceID = getWorkspaceID();
+
     @inject(GitpodInfoService) protected infoProvider: GitpodInfoService;
-    @inject(GitpodServiceProvider) protected serviceProvider: GitpodServiceProvider;
     @inject(OpenerService) protected readonly openerService: OpenerService;
     @inject(ApplicationShell) protected readonly shell: ApplicationShell;
     @inject(MiniBrowserOpenHandler) protected miniBrowserOpenHandler: MiniBrowserOpenHandler;
     @inject(FrontendApplicationStateService) protected applicationStateService: FrontendApplicationStateService;
     @inject(GitpodGitTokenProvider) protected gitTokenProvider: GitpodGitTokenProvider;
     @inject(EditorManager) protected readonly editorManager: EditorManager;
 
+    private readonly isWorkspaceOwner: Promise<boolean>;
 
-    protected isWorkspaceOwner = new Deferred<boolean>();
-    protected gitpodInfo: GitpodInfo;
```

Comment:
it was never safe to assume that `gitpodInfo` cannot be  `undefined`, worked by luck.